src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -417,20 +417,12 @@ public async ValueTask ApplyAsync(IVsAsyncFileChangeEx service, CancellationToke
 
                     case Kind.UnwatchDirectories:
                         Contract.ThrowIfFalse(_cookies is not null);
-                        foreach (var unwatchCookie in _cookies)
-                        {
-                            await service.UnadviseDirChangeAsync(unwatchCookie, cancellationToken).ConfigureAwait(false);
-                        }
-
+                        await service.UnadviseDirChangesAsync(_cookies, cancellationToken).ConfigureAwait(false);
                         return;
 
                     case Kind.UnwatchFiles:
                         Contract.ThrowIfFalse(_tokens is not null);
-                        foreach (var token in _tokens)
-                        {
-                            await service.UnadviseFileChangeAsync(token.Cookie!.Value, cancellationToken).ConfigureAwait(false);
-                        }
-
+                        await service.UnadviseFileChangesAsync(_tokens.Select(token => token.Cookie!.Value).ToArray(), cancellationToken).ConfigureAwait(false);
```

Comment:
Do we have a SelectAsArray() that might be more efficient here? We have some overloads that try to limit allocations but no sense if they're useful here.

Reply:
➡️ We don't have anything obviously better for source type `IEnumerable<T>`. It also doesn't seem like it would be observably different for this call.